src/main/java/seedu/duke/storage/TaskList.java

```diff
@@ -0,0 +1,103 @@
+package seedu.duke.storage;
+
+import java.util.ArrayList;
+
+import seedu.duke.commands.Ui;
+import seedu.duke.exceptions.action.DukeActionOutOfBoundException;
+import seedu.duke.tasks.Task;
+
+public class TaskList {
+    private final ArrayList<Task> taskList;
+
+    /**
+     * Primary Constructor.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Retrieves the list of tasks that is being stored in this class.
+     * 
+     * @return an ArrayList of tasks which is being stored.
+     * @see java.util.ArrayList
+     */
+    public ArrayList<Task> getTaskList() {
+        return this.taskList;
+    }
+
+    /**
+     * Loads previous saved Tasks from a txt file. Save it in this current TaskList
+     * in an ArrayList of Task object.
+     * 
+     * @param savedTasks this is the ArrayList of Task coming from the txt file.
+     * @see java.util.ArrayList
+     */
+    public void loadFromStorage(ArrayList<Task> savedTasks) {
+        this.taskList.addAll(savedTasks);
+    }
+
+    /**
+     * Helps to mark the task in the list of Tasks as done.
+     * 
+     * @param number is the Task id.
+     */
+    public void doneItem(String number) {
+        try {
+            int index = Integer.parseInt(number) - 1;
+            Task curr = this.taskList.get(index);
+            this.taskList.set(index, curr.markAsDone());
+            curr = this.taskList.get(index);
+
+            Ui.printMessage(Ui.DONE_MESSAGE + "\n" + Ui.INDENT + "  " + curr.toString());
```

Comment:
In terms of separation of concerns, would it be better to place this Ui related part of the program outside of this class, which handles storage-specific tasks?